mempool/mempool.go

```diff
@@ -677,16 +708,28 @@ func (cache *mapTxCache) Reset() {
 	cache.mtx.Unlock()
 }
 
-// Push adds the given tx to the cache and returns true. It returns false if tx
+// PushTxWithInfo adds the given tx to the cache and returns true. It returns false if tx
 // is already in the cache.
-func (cache *mapTxCache) Push(tx types.Tx) bool {
+func (cache *mapTxCache) PushTxWithInfo(tx types.Tx, txInfo TxInfo) bool {
 	cache.mtx.Lock()
 	defer cache.mtx.Unlock()
 
 	// Use the tx hash in the cache
 	txHash := sha256.Sum256(tx)
-	if moved, exists := cache.map_[txHash]; exists {
-		cache.list.MoveToBack(moved)
+	if listEntry, exists := cache.map_[txHash]; exists {
```

Comment:
The commented code is essential for this PR as without this functionality we only remember single sender per tx. So in case we receive tx from N peers we will forward it to N-1 peer. in case it's too complex adding senders list to main mempool list, what if we would keep senders as part of cache? As senders list is relevant only for tx that are in main mempool list we can empty senders list when when we remove tx from the main mempool list.